.github/workflows/production-heartbeat.yml

```diff
@@ -8,136 +8,7 @@ on:
-
-      # === Attempt to execute the smoke tests ===
-      - name: Run smoke tests
-        id: smoke_tests
-        run: smoke-tests/smoke-test${{ matrix.os.exe }} sf
-
-      # === Upload the smoke-test-results folder as an artifact ===
-      - name: Upload smoke-test-results folder as artifact
-        if: ${{ always() }}
-        uses: actions/upload-artifact@v4
-        with:
-          name: smoke-test-results-${{ runner.os }}
-          path: smoke-test-results
-
-      # === Report any problems ===
-      - name: Report problems
-        # There are problems if any step failed or was skipped.
-        # Note that the `join()` call omits null values, so if any steps were skipped, they won't have a corresponding
-        # value in the string.
-        if: ${{ failure() || cancelled() }}
-        shell: bash
-        env:
-          # Build the status strings for each step as environment variables to save space later. Null retry_count values
-          # will be replaced with `n/a` to maintain readability in the alert.
-          CLI_INSTALL_STATUS: ${{ steps.sf_install.outcome }} after ${{ steps.sf_install.outputs.retry_count || 'n/a' }} retries
-          SCANNER_INSTALL_STATUS: ${{ steps.scanner_install.outcome }} after ${{ steps.scanner_install.outputs.retry_count || 'n/a' }} retries
-          SMOKE_TESTS_STATUS: ${{ steps.smoke_tests.outcome }}
-          # A link to this run, so the PagerDuty assignee can quickly get here.
-          RUN_LINK: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
-        run: |
-          # GHA env-vars don't have robust conditional logic, so we'll use this if-else branch to define some bash env-vars.
-          ALERT_SEV="critical"
-          ALERT_SUMMARY="Production heartbeat script failed on ${{ runner.os }}"
-          # Define a helper function to create our POST request's data, to sidestep issues with nested quotations.
-          generate_post_data() {
-          # This is known as a HereDoc, and it lets us declare multi-line input ending when the specified limit string,
-          # in this case EOF, is encountered.
-          cat <<EOF
-          {"payload": {
-            "summary": "${ALERT_SUMMARY}",
-            "source": "Github Actions",
-            "severity": "${ALERT_SEV}",
-            "custom_details": "SF install: ${{ env.CLI_INSTALL_STATUS }}. Scanner install: ${{ env.SCANNER_INSTALL_STATUS }}. Smoke tests: ${{ env.SMOKE_TESTS_STATUS }}."
-          },
-          "links": [{
-            "href": "${{ env.RUN_LINK }}",
-            "text": "Link to action execution"
-          }],
-          "event_action": "trigger",
-          "dedup_key": "GH-HB-${{ matrix.os.vm }}-${{ matrix.node }}",
-          "routing_key": "${{ secrets.PAGERDUTY_HEARTBEAT_KEY }}"
-          }
-          EOF
-          }
-          # Make our POST request
-          curl --request POST --data "$(generate_post_data)" https://events.pagerduty.com/v2/enqueue
+jobs:
```

Comment:
This is technically unnecessary, since un-defaulting `dev-4` will disable the cron job that causes this workflow to run. But I'm doing it anyway to be thorough, and because we might need to make v4 the default branch again in the future or something like that, and it's worth being prepared.